build/lib/compilation.ts

```diff
@@ -47,24 +47,31 @@ interface ICompileTaskOptions {
 	readonly emitError: boolean;
 	readonly transpileOnly: boolean | { esbuild: boolean };
 	readonly preserveEnglish: boolean;
+	readonly noEmit?: boolean;
+	readonly logTopic?: string;
 }
 
-export function createCompile(src: string, { build, emitError, transpileOnly, preserveEnglish }: ICompileTaskOptions) {
+export function createCompile(src: string, { build, emitError, transpileOnly, preserveEnglish, noEmit, logTopic }: ICompileTaskOptions) {
 	const tsb = require('./tsb') as typeof import('./tsb');
 	const sourcemaps = require('gulp-sourcemaps') as typeof import('gulp-sourcemaps');
+	const currentReporter = logTopic ? createReporter(logTopic) : reporter;
 
 
 	const projectPath = path.join(__dirname, '../../', src, 'tsconfig.json');
 	const overrideOptions = { ...getTypeScriptCompilerOptions(src), inlineSources: Boolean(build) };
+	if (noEmit) {
+		overrideOptions.noEmit = true;
```

Comment:
I didn't test this yet but doesn't `noEmit` also mean no `.d.ts` files are created and therefore the impact of a code change cannot be assessed correctly anymore? E.g did its d.ts change and hence more checking is need? 

I think we need to check the `noEmit` flag where we send the JS files into the stream